### PR TITLE
Switch wasmi to wasmer

### LINKED ISSUE #541: Switch wasmi to wasmer
wasmer is currently 100X faster than wasmi. So it would give us huge boost in performance.

It's still in development and might not be as stable on all platforms as wasmi.

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
It might be unstable on linux according to https://spectrum.chat/wasmer/runtime/interrupting-execution~2e0b4dbd-757b-4326-a0de-ab8ef1a46e05?m=MTU0OTY3MzIzMTg3Mg==

But works great on mac. Need more testing on linux.

Fixes #541

### CODE REVIEW

File: core/wasm/src/executor.rs [hunk 1]
```diff
@@ -30,66 +36,60 @@ pub fn execute<'a>(
...
-    let instantiation_resolver = EnvModuleResolver::with_memory(memory.clone());
-    // Make a module instance
-    let module_instance = wasmi::ModuleInstance::new(
-        &module,
-        &wasmi::ImportsBuilder::new().with_resolver("env", &instantiation_resolver),
-    ).map_err(Error::Interpreter)?;
+    let instrumented_code = prepare::prepare_contract(code, &config).map_err(Error::Prepare)?;
+
+    let memory = Memory::new(MemoryDescriptor {
+        minimum: Pages(17),
```
Comment (bowenwang1996) [inline core/wasm/src/executor.rs]: How are these numbers determined?

File: core/wasm/src/executor.rs [hunk 2]
```diff
@@ -30,66 +36,60 @@ pub fn execute<'a>(
...
 
     // All public functions should start with `PUBLIC_FUNCTION_PREFIX` in WASM.
-    let method_name = format!("{}{}", PUBLIC_FUNCTION_PREFIX, std::str::from_utf8(method_name).map_err(|_| Error::BadUtf8)?);
+    let method_name = format!("{}{}",
+        PUBLIC_FUNCTION_PREFIX,
+        std::str::from_utf8(method_name).map_err(|_| Error::BadUtf8)?);
 
-    match module_instance.run_start(&mut runtime) {
+    // Resolving function by method_name
+    // let mut func = instance.func(&method_name)
```
Comment (bowenwang1996) [inline core/wasm/src/executor.rs]: ?

File: core/wasm/src/prepare.rs [hunk 3]
```diff
@@ -218,34 +212,20 @@ pub(super) fn prepare_contract(
                 // to configured maximum.
                 return Err(Error::Memory);
             }
-            (initial, maximum) => Memory::init(initial, maximum),
+            // (initial, maximum) => Memory::init(initial, maximum),
```
Comment (bowenwang1996) [inline core/wasm/src/prepare.rs]: ?

**Comment by evgenykuzyakov** [comment]
@bowenwang1996 @azban 
Is there a way to filter some debug logs? 
cranelift spams with things like:
[2019-02-09T19:17:49Z INFO  cranelift_wasm::func_translator] translate(198 bytes, u0:0(i32, i64 vmctx) -> i32 system_v)
[2019-02-09T19:17:49Z INFO  cranelift_wasm::func_translator] translate(36 bytes, u0:1(i32, i64 vmctx) -> i32 system_v)
[2019-02-09T19:17:49Z INFO  cranelift_wasm::func_translator] translate(61 bytes, u0:2(i32, i32, i64 vmctx) -> i32 system_v)
[2019-02-09T19:17:49Z INFO  cranelift_wasm::func_translator] translate(115 bytes, u0:3(i32, i32, i32, i32, i32, i64 vmctx) -> i32 system_v)
[2019-02-09T19:17:49Z INFO  cranelift_wasm::func_translator] translate(122 bytes, u0:4(i32, i32, i64 vmctx) -> i32 system_v)
[2019-02-09T19:17:49Z INFO  cranelift_wasm::func_translator] translate(113 bytes, u0:5(i32, i64 vmctx) -> i32 system_v)

**Comment by bowenwang1996** [comment]
You can change which targets to keep when initiating the logger.

**Comment by azban** [comment]
@evgenykuzyakov you'll want to set up a filter for cranelift_wasm target and set level to warn

**Pull request merged**